symfony/plugins/orangehrmRESTPlugin/lib/Api/User/EmployeePunchOutAPI.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace Orangehrm\Rest\Api\User;
+
+use Orangehrm\Rest\Api\Exception\InvalidParamException;
+use Orangehrm\Rest\Api\Exception\RecordNotFoundException;
+use Orangehrm\Rest\Api\Exception\BadRequestException;
+use Orangehrm\Rest\Http\Response;
+use Orangehrm\Rest\Api\Attendance\PunchOutAPI;
+use \PluginAttendanceRecord;
+use \DateTimeZone;
+use \DateTime;
+use \Exception;
+
+
+class EmployeePunchOutAPI extends PunchOutAPI
+{
+    /**
+     * @return Response
+     * @throws InvalidParamException
+     * @throws RecordNotFoundException
+     */
+    public function savePunchOut()
+    {
+        $filters = $this->filterParameters();
+        $timeZone = $filters['timeZone'];
+        $punchOutNote = $filters['punchOutNote'];
+        $dateTime = $filters['dateTime'];
+        $editable = $this->getAttendanceService()->getDateTimeEditable();
+        if ($editable && empty($dateTime)) {
```

Comment:
This logic can be moved into the pre-validation level and reject the request then and there.